packages/database/src/modelClasses/MapOverlayGroupRelation.js

```diff
@@ -17,6 +18,26 @@ const RELATION_CHILD_TYPES = {
 export class MapOverlayGroupRelationRecord extends DatabaseRecord {
   static databaseRecord = RECORDS.MAP_OVERLAY_GROUP_RELATION;
 
+  static joins = [
+    {
+      joinType: JOIN_TYPES.LEFT,
+      joinWith: RECORDS.MAP_OVERLAY,
+      joinAs: 'child_map_overlay',
+      joinCondition: [`${RECORDS.MAP_OVERLAY_GROUP_RELATION}.child_id`, `child_map_overlay.id`],
+      fields: { code: 'code' },
+    },
+    {
+      joinType: JOIN_TYPES.LEFT,
+      joinWith: RECORDS.MAP_OVERLAY_GROUP,
+      joinAs: 'child_map_overlay_group',
+      joinCondition: [
+        `${RECORDS.MAP_OVERLAY_GROUP_RELATION}.child_id`,
+        `child_map_overlay_group.id`,
+      ],
+      fields: { code: 'code' },
+    },
+  ];
```

Comment:
I suspect this is what’s causing the **central-server** tests to fail, and it might be indicative of something bigger that isn’t currently a problem, but could be difficult to debug for a future dev

There’s a unit test which makes this query:

```sql
SELECT "map_overlay_group_relation"."id",
       "map_overlay_group_relation"."map_overlay_group_id",
       "map_overlay_group_relation"."child_id",
       "map_overlay_group_relation"."child_type",
       "map_overlay_group_relation"."sort_order",
       "child_map_overlay"."code" AS "code",
       "child_map_overlay_group"."code" AS "code"
FROM "map_overlay_group_relation"
     LEFT JOIN "map_overlay" AS "child_map_overlay" 
               ON "map_overlay_group_relation"."child_id" = "child_map_overlay"."id"
     LEFT JOIN "map_overlay_group" AS "child_map_overlay_group"
               ON "map_overlay_group_relation"."child_id" = "child_map_overlay_group"."id"
WHERE "id" = $1
LIMIT $2
```

The immediate problem which is causing the test failure is that `id` in the `WHERE` clause is ambiguous, where we presumably want `"map_overlay_group_relation"."id"`. However, `code` would be equally ambiguous.

Any ideas about how to make these fully qualified?

Reply:
Taking a look at the `DatabaseModel` class, it seems like we don't attempt to fully qualify the fields in the where clause (we do in the select clause tho). Eg. `DatabaseModel#find()`

We actually do seem to have some logic around this with `DatabaseModel#findById()` function, as it uses `getIdClause()` that does fully qualify the `id` field. However, I believe the line in the tests that causes this this issue is `setupMapOverlayTestData#findOrCreateMapOverlayGroup()` which calls `DatabaseModel#find()` underneath.

I've just pushed a commit that takes a stab at using fully qualified names in the where clause, hopefully that resolves the issue without breaking other things!